client/src/components/widgets/PlotlyPlot/script.js

```diff
@@ -199,55 +261,69 @@ export default {
       let xAxis = image.layout.xaxis.title.text;
       this.$store.commit(`${this.itemId}/PLOT_X_AXIS_SET`, xAxis);
     },
+    setPlotLabels(image) {
+      this.plotLabels = {
+        title: image.layout.title.text,
+        xaxis: image.layout.xaxis.title.text,
+        yaxis: image.layout.yaxis.title.text,
+      };
+    },
     applyLogScaling(image) {
       image.layout.xaxis.type = this.logScaling ? "log" : "linear";
       image.layout.yaxis.type = this.logScaling ? "log" : "linear";
     },
     applyZoom(image) {
-      image.layout.xaxis.range = this.zoom.xAxis;
-      image.layout.yaxis.range = this.zoom.yAxis;
+      image.layout.xaxis.range = this.zoom.xaxis;
+      image.layout.yaxis.range = this.zoom.yaxis;
       image.layout.yaxis.autorange = false;
     },
     useGlobalRange(image) {
       image.layout.yaxis.range = [...this.range];
       image.layout.yaxis.autorange = false;
     },
+    useRunGlobals(image) {
+      if (this.logScaling || this.zoom) {
+        return;
+      }
+
+      image.layout.xaxis.range = [...this.xRange];
+      image.layout.xaxis.autorange = false;
+      image.layout.yaxis.range = [...this.yRange];
+      image.layout.yaxis.autorange = false;
+    },
+    updatePlotDetails(image) {
+      ["xaxis", "yaxis"].forEach((axis) => {
+        image.layout[`${axis}`].title.text = this[`${axis}Visible`]
+          ? this.plotLabels[`${axis}`]
+          : "";
+        image.layout[`${axis}`].showticklabels = this[`${axis}Visible`];
+      });
+      image.layout.title.text = this.showTitle ? this.plotLabels.title : "";
+      image.layout.margin.b = this.xaxisVisible ? 30 : 10;
+      image.layout.margin.l = this.yaxisVisible ? 60 : 10;
+      image.layout.margin.t = this.showTitle ? 30 : 10;
+    },
     plotPreProcessing(image) {
       if (!this.xAxis) this.setXAxis(image);
+      if (!this.plotLabels.title) this.setPlotLabels(image);
       this.applyLogScaling(image);
       image.layout.yaxis.autorange = true;
       image.layout.showlegend = this.legendVisibility;
-      if (this.zoom) this.applyZoom(image);
       if (this.range) this.useGlobalRange(image);
+      if (this.runGlobals) this.useRunGlobals(image);
+      if (this.zoom) this.applyZoom(image);
+      const data = image.data[0];
+      const xRange = [Math.min(...data.x), Math.max(...data.y)];
```

Comment:
Those Math.min(...array) are not great. Might be worth to have a utility function that do it efficiently.

```javascript
function extractRange(array) {
   let minValue = array[0];
   let maxValue = array[0];
   for (let i = 0; i < array.length; i++) {
       const v = array[i];
       if (v < minValue) {
            minValue = v;
       } else if (v > maxValue) {
            maxValue = v;
       }
   }
   return [minValue, maxValue]
}
```

Reply:
Understood, thank you! I've updated the instances of min/max and the spread operator to use the helper function instead.